### PR TITLE
Internal: clean error handling of cmdline parsing

### LINKED ISSUE #1139: Bug: fatal errors are swallowed
**Describe the bug**
Since https://github.com/ocaml-ppx/ocamlformat/pull/1127 fatal errors such as `Fatal error: exception ("User Error: malformed .ocamlformat file"` that used to `exit 2` with an uncaught exception now just silently succeed without doing anything.

### PR DESCRIPTION
**Pull request by gpetiot**
Cleaning when investigating #1139

### CODE REVIEW

**Comment by gpetiot** [comment]
Rebased, and grouped together with the other "CLI refactoring" PRs in the changelog.

File: lib/Conf.mli [hunk 1]
```diff
@@ -107,7 +107,7 @@ type opts =
         (** Check whether the formatted output exceeds the margin. *) }
 (** Options changing the tool's behavior *)
 
-val action : unit -> action * opts
+val action : unit -> action Cmdliner.Term.result * opts
```
Comment (Julow) [inline lib/Conf.mli]: This should be `(action * opts) Cmdliner.Term.result`. Eg. constructing `opts` in `validate` in lib/Conf.ml

File: bin/ocamlformat.ml [hunk 2]
```diff
@@ -75,7 +75,7 @@ let action, opts = Conf.action ()
 
 ;;
 match action with
-| Inplace inputs ->
+| `Ok (Inplace inputs) ->
```
Comment (Julow) [inline bin/ocamlformat.ml]: What do you think of moving the action cases to a `action -> int` function, returning the exit status ?

Comment (gpetiot) [reply bin/ocamlformat.ml]: I did something slightly different, returning a result type instead, it's better I think.

File: lib/Conf.mli (hunk 1 above)
Comment (gpetiot) [reply lib/Conf.mli]: done

**Pull request merged**